utils/regr/regrmysql.cpp

```diff
@@ -1494,6 +1502,8 @@ extern "C"
             long double sumy = data->sumy;
             long double sumxy = data->sumxy;
             long double covar_samp = (sumxy - ((sumx * sumy) / N)) / (N - 1);
+            if (covar_samp < 0)  // might be -0
+                covar_samp = 0;
```

Comment:
The above block is repeated in so many places. It would be nice to make a function or a class and reuse it everywhere.

Reply:
Yeah. I thought about a sub-class and function, but decided I didn't want to spend the cpu cycles for the call. Could be done easily enough.

Reply:
In this case covar_samp could have a legitimate negative value, so I removed the check